src/buildifier/buildifier.ts

```diff
@@ -126,14 +126,26 @@ export function getBuildifierFileType(fsPath: string): BuildifierFileType {
   // https://github.com/bazelbuild/buildtools/blob/d39e4d/build/lex.go#L88)
   // so that user actions in the IDE are consistent with the behavior they
   // would see running buildifier on the command line.
-  const parsedPath = path.parse(fsPath.toLowerCase());
+  const raw = fsPath.toLowerCase();
+  let parsedPath = path.parse(raw);
+  if (parsedPath.ext === ".oss") {
+    parsedPath = path.parse(parsedPath.name);
+  }
   if (parsedPath.ext === ".bzl" || parsedPath.ext === ".sky") {
     return "bzl";
   }
-  if (parsedPath.ext === ".build" || parsedPath.base === "build") {
+  if (
+    parsedPath.ext === ".build" ||
+    parsedPath.name === "build" ||
+    parsedPath.name.startsWith("build")
```

Comment:
You've dropped the dot from the end of this check (and below, with `workspace`). The original Go check is [`strings.HasPrefix(base, "build.")`](https://github.com/bazelbuild/buildtools/blob/master/build/lex.go#L125); presumably this lets them catch files named like "BUILD.foo.bar" where stripping the extension only leaves "BUILD.foo".

Without the dot, this will also match things like "buildfoo", which could lead to subtle differences.

Reply:
Oh yep! Thanks for your patience, fixin...